### ISSUE TITLE
[PyTorch Backend] Triton server consumes too much GPU memory

### ISSUE THREAD

**Issue by KeeeeiZ** [triton-inference-server/server#3265]
**Description**
I have model trained with pytorch 1.9. When I load the model with `jit.load` locally and use it to compute a bunch of data, the model only consumed about 4Gb of GPU memory, but when I put the model int triton server and send the same data to the server to inference, it consumed more than 15Gb of GPU memory.

**Triton Information**
I use docker image with version 20.10-py3

Are you using the Triton container or did you build it yourself?

**To Reproduce**
Steps to reproduce the behavior.

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

the config is 
```
name: "model_name"
platform: "pytorch_libtorch"
max_batch_size: 128

input [
  {
   name: "INPUT__0"
   data_type: TYPE_INT64
   dims: [-1 ]
  },
  {
   name: "INPUT__1"
   data_type: TYPE_INT64
   dims: [-1 ]
  },
  {
   name:"INPUT__2"
   data_type:TYPE_INT64
   dims: [-1 ]
  }
]

output [
  {
    name: "OUTPUT__0"
    data_type: TYPE_FP32
    dims: [ -1]
  }
]

instance_group [
  {
    count: 1
    gpus: 0
    kind: KIND_GPU
  }
]

version_policy: { specific { versions: [ 1 ] }}
```

The code I use to request is 

```python
    inputs = []
    outputs = []
    inputs.append(httpclient.InferInput('INPUT__0', [input2_data.shape[0], input2_data.shape[1]], "INT64"))
    inputs.append(httpclient.InferInput('INPUT__1', [input2_data.shape[0], input2_data.shape[1]], "INT64"))
    inputs.append(httpclient.InferInput('INPUT__2', [input2_data.shape[0], input2_data.shape[1]], "INT64"))
    # Initialize the data
    inputs[0].set_data_from_numpy(input0_data.numpy(), binary_data=False)
    inputs[1].set_data_from_numpy(input1_data.numpy(), binary_data=False)
    inputs[2].set_data_from_numpy(input2_data.numpy(), binary_data=False)

    outputs.append(httpclient.InferRequestedOutput('OUTPUT__0', binary_data=False))

    results = triton_client.infer(
        'remove_extra_line_space',
        inputs,
        outputs=outputs,
    )
```

**Expected behavior**
It should not consume so much more GPU memory than running the model locally.

**Comment by CoderHam** [comment]
@KeeeeiZ which version of pytorch did you use to run your experiments? Was it the same as in 20.10-py3?
Please refer to the [Framework Support Matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html#unique_1997996242).

**Comment by KeeeeiZ** [comment]
> @KeeeeiZ which version of pytorch did you use to run your experiments? Was it the same as in 20.10-py3?
> Please refer to the [Framework Support Matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html#unique_1997996242).

The matrix shows 20.10 docker image has a `1.7` version of pytorch while my experiment is run under pytorch 1.9.

**Comment by KeeeeiZ** [comment]
Plus, the model is trained with pytorch 1.7.

**Comment by vancuonghoang** [comment]
I had the same problem when I tried with my tf2 model and consumed ~ 13Gb of GPU memory, so I converted my model and used ONNX backend, after that problem solved (now it just take 1.7 ~2 GB VRAM) . 
I think maybe you should try!

**Comment by dyastremsky** [comment]
Closing due to inactivity.
